proto/identity/associations/association.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+
+package xmtp.identity.associations;
+
+import "identity/associations/signature.proto";
+
+option go_package = "github.com/xmtp/proto/v3/go/identity/associations";
+option java_package = "org.xmtp.proto.identity.associations";
+
+// The first entry of any XID log, which is considered valid iff the XID can be
+// deterministically derived from the address and nonce.
+// The recovery address defaults to the initial_associated_address unless
+// there is a subsequent ChangeRecoveryAddress in the log.
+message CreateXid {
+  string xid = 1;
+  uint32 nonce = 2;
+  string initial_associated_address = 3;
+}
+
+message AddAssociation {
+  enum AssociatedRole {
+    ASSOCIATED_ROLE_UNSPECIFIED = 0;
+    ASSOCIATED_ROLE_INSTALLATION = 1;
+    ASSOCIATED_ROLE_ADDRESS = 2;
+    ASSOCIATED_ROLE_LEGACY_KEY = 3;
+  }
+
+  string xid = 1;
+  uint64 client_timestamp_ns = 2;
+  uint32 nonce = 3;
+  Signature existing_member_signature = 4;
+  Signature new_member_signature = 5;
+  AssociatedRole new_member_role = 6;
+}
+
+// An entity may have been associated with multiple roles. Clients MUST remove
+// the entity from all previously associated roles.
+message RevokeAssociation {
+  string xid = 1;
+  uint64 client_timestamp_ns = 2;
+  uint32 nonce = 3;
+  Signature recovery_address_signature = 4;
+  // The nonce of the association being removed.
+  // This nonce must never be used again for this member.
+  uint32 revoked_nonce = 4;
+  // Either an installation public key or a wallet address
+  string revoked_member = 5;
+  // Allow the revoker to include a list of downstream associations that should
+  // be retained even after the revocation. Any association not in this list
+  // will be removed
+  repeated string retained_associations = 6;
```

Comment:
What are these strings? I think a hash of each association makes the most sense here.

Reply:
I was imagining the IDs of the entities (wallet address or installation pub key), but the hash of the association would be fine too.

Reply:
Removed these as the design no longer needs them